### DISCUSSION TITLE
Same built-in delegates for Windows and Linux builds

### DISCUSSION THREAD

**Discussion by williamjacobszoon** [ImageMagick/ImageMagick#4115] [Development] [answered]
Is it possible to have the same built-in delegates configured in both Windows and Linux builds?

```
Version: ImageMagick 7.1.0-5 Q16 x86_64 2021-08-14 https://imagemagick.org
Copyright: (C) 1999-2021 ImageMagick Studio LLC
License: https://imagemagick.org/script/license.php
Features: Cipher DPC HDRI OpenMP(4.5)
Delegates (built-in): bzlib djvu fontconfig freetype jbig jng jpeg lcms lqr lzma openexr png raqm tiff webp x xml zlib
```

```
Version: ImageMagick 7.1.0-4 Q16 x64 2021-07-18 https://imagemagick.org
Copyright: (C) 1999-2021 ImageMagick Studio LLC
License: https://imagemagick.org/script/license.php
Visual C++: 192930038
Features: Cipher DPC HDRI OpenCL
Delegates (built-in): bzlib cairo flif freetype gslib heic jng jp2 jpeg jxl lcms lqr lzma openexr pangocairo png ps raqm raw rsvg tiff webp xml zip zlib
```

**Comment by dlemstra** [comment, marked as answer]
Not every delegate that is available on Linux works on Windows. So the short answer is maybe.It looks like the bottom one is Windows and has more delegates. If you want to also have those delegates on Linux you will need to rebuild ImageMagick with those delegates installed on the machine. What is your end goal and why are you asking this question?

**Reply by williamjacobszoon** [reply]
It's more around the consistency between builds. I think its better to have the standard releases with the same delegates on both Linux and Windows builds. Also, I think this gives the assurance when we deploy applications between Linux and windows which uses ImageMagick in the command-line mode will work in the same way on both platforms.

**Reply by dlemstra** [reply]
If you need to have the same delegates on every machine you will need to resolve that yourself.
